src/Compile-SourceScript/Public/Compile-SourceScript.ps1

```diff
@@ -145,11 +146,48 @@ function Compile-SourceScript {
                     $sourceFile.Name
                 )
             }
+            $returnExitCode = $false
+            if ($MOD_NAME -eq 'sourcemod') {
+                $pluginErrorRegexPattern = '^.*\.sp\(\d+\)\s*:\s*error (\d+)'
+                if (!$PSBoundParameters['SkipWrapper'] -or $env:OS -ne 'Windows_NT') {
+                    $returnExitCode = $true
+                }
+            }elseif ($MOD_NAME -eq 'amxmodx') {
+                $pluginErrorRegexPattern = '^.*\.sma\(\d+\)\s*:\s*error (\d+)'
+                $returnExitCode = $true
+            }
             New-Item -Path $COMPILED_DIR -ItemType Directory -Force | Out-Null
 
             # Begin compilation
             if ($PSBoundParameters['SkipWrapper']) { "Compiling $($sourceFile.Name)..." | Write-Host -ForegroundColor Yellow }
-            Start-Process @processArgs
+
+            if ($returnExitCode) {
+                # The amxmodx compiler always exits with exit code 0, so we have to use a regex on the stdout
+                $global:LASTEXITCODE = 0
+
+                $tempDir = Join-Path ([System.IO.Path]::GetTempPath()) (New-Guid).Guid
+                $stdoutFile = Join-Path $tempDir 'stdout'
+                $stderrFile = Join-Path $tempDir 'stderr'
+                $processArgs['RedirectStandardOutput'] = $stdoutFile
+                $processArgs['RedirectStandardError'] = $stderrFile
+                New-Item $tempDir -ItemType Directory -Force > $null
+                $p = Start-Process @processArgs
+                $stdout = Get-Content $stdoutFile
+                $stdout | Write-Host
+                $stderr = Get-Content $stderrFile
```

Comment:
The variable `$stderr` is never used. Based on my testing the variable is always empty together with the stderr file. I believe the compiler and compiler wrappers only use stdout. In which case it may be better to not use redirection to stdout and stderr files, since redirection also means the inability to display output in the console in real-time and in the correct order. I added outputting of stdout as a workaround, but adding outputting of stderr would be hackerish leading to incorrect order of corresponding stdout and stderr lines for each instance of compilation as in the case when multiple scripts are compiled when the compiler wrappers are used. Because of this reason I think it would be better if redirection of stdout and stderr to files aren't used, which would be possible if we are confident that the compilers and compiler wrappers only output to stdout.

Reply:
We can use `Tee-Object $stdoutFile | Write-Host` write to file and print to console @joeltimothyoh

Reply:
That would mean not using redirect params for `Start-Process`. Tee-Object is perfect for if we’re confident stdout is only used by compiler and compiler wrappers as I mentioned.

Reply:
@joeltimothyoh based on the logs, I don't see any stderr output emitted by the compiler or the compiler wrapper. I think we can be confident to rely solely on regex patterns on the stdout, which will work consistently in detecting compilation errors regardless of whether compilation is done via compiler or the compiler wrapper.

Reply:
To capture `stdout` of `Start-Process` is not as easy as `Start-Process .. | Tee-Object`. It's tricky. See https://stackoverflow.com/questions/8761888/capturing-standard-out-and-error-with-start-process and http://www.blackwasp.co.uk/CaptureProcessOutput.aspx. I suggest we stick with `Start-Process -RedirectStandardOutput x -RedirectStandardError y`. @joeltimothyoh I'm pushing changes

Reply:
```powershell
$p = start-process -FilePath powershell -ArgumentList '-c', 'foreach ($i in (1..5)){ sleep 1; echo hi; }' -NoNewWindow  -Wait -PassThru -RedirectStandardOutput ~/stdout.log -RedirectStandardError ~/stderr.log
```

Once `-RedirectStandardOutput` and `-RedirectStandardError y` are used then `Start-Process` will not print `stdout` and `stderror` to the console. So that might be a little impairing for the developer who might want to see realtime compilation progress.

Since compilation is quick it should not take too long before compilation would be completed and printed to console, so I think this should be fine for the developer.

Thoughts @joeltimothyoh

Reply:
> Once -RedirectStandardOutput and -RedirectStandardError y are used then Start-Process will not print stdout and stderror to the console. So that might be a little impairing for the developer who might want to see realtime compilation progress.

This is the very issue I've already explained above in https://github.com/startersclan/Compile-SourceScript/pull/7#discussion_r468039125 and https://github.com/startersclan/Compile-SourceScript/pull/7#discussion_r468095470. Best to not redirect any output to files if possible; that makes using Tee-Object easy with Start-Process, and lessen unnecessary moving parts of the module.

Reply:
@joeltimothyoh the problem is that there is no way of capturing the stdout returned by `Start-Process`, as i explained above, see https://github.com/startersclan/Compile-SourceScript/pull/7#discussion_r468238198. 

I agree the best is to preserve console output, but it's not possible without using `Invoke-Process` as suggested in the second link.

Reply:
> @joeltimothyoh the problem is that there is no way of capturing the stdout returned by Start-Process, as i explained above, see #7 (comment).

I'll have to debug to find out.

> I agree the best is to preserve console output, but it's not possible without using Invoke-Process as suggested in the second link.

`Invoke-Process` as in https://www.powershellgallery.com/packages/Invoke-Process/1.4? If possible it'd be better to stick to in-built cmdlets.

Reply:
> @joeltimothyoh the problem is that there is no way of capturing the stdout returned by Start-Process, as i explained above, see #7 (comment).

I've tested and this seems to be correct.

> Since compilation is quick it should not take too long before compilation would be completed and printed to console, so I think this should be fine for the developer.

I think so too. The order of the output should be fine so long as the compiler doesn't output to stderr.

Reply:
Fixed. See related discussion in https://github.com/startersclan/Compile-SourceScript/pull/7/files/eff914f808b9f4883e8d064020bdc8422ac214e8#r479858193